server/src/main/scala/org/scassandra/server/cqlmessages/request/ExecuteRequest.scala

```diff
@@ -46,7 +46,7 @@ object ExecuteRequest {
     val numberOfVariables = bodyIterator.getShort
 
     val variableValues = variableTypes.map (varType => {
-      varType.readValue(bodyIterator, VersionTwo)
+      varType.readValueWithLength(bodyIterator)(VersionTwo)
```

Comment:
Reminding me of the nastyness I did with methods for each version rather than use the CqlMessageFactory :(